examples/md-flexible/src/ParallelVtkWriter.cpp

```diff
@@ -128,6 +153,93 @@ void ParallelVtkWriter::recordTimestep(const int &currentIteration,
   timestepFile.close();
 }
 
+void ParallelVtkWriter::recordDomainSubdivision(const int &currentIteration,
+                                                const autopas::Configuration &autoPasConfiguration,
+                                                const RegularGridDecomposition &decomposition) {
+  if (_mpiRank == 0) {
+    createPvtsFile(currentIteration, decomposition);
+  }
+
+  std::ostringstream timestepFileName;
+  timestepFileName << _dataFolderPath << _sessionName << "_subdivision_" << _mpiRank << "_" << std::setfill('0')
+                   << std::setw(_maximumNumberOfDigitsInIteration) << currentIteration << ".vts";
+
+  std::ofstream timestepFile;
+  timestepFile.open(timestepFileName.str(), std::ios::out | std::ios::binary);
```

Comment:
Ok I tested a bit around and this is the way to read / write binary data:

```cpp
  // define data
  std::string filename{"theFile.out"};
  size_t payload{0x42};
  
  // write binary
  auto leFileWrite = std::fstream(filename, std::ios::out);
  leFileWrite.write(reinterpret_cast<const char *>(&payload), sizeof(payload));
  leFileWrite.close();

  // read data
  auto leFileRead = std::fstream(filename, std::ios::in);
  decltype(payload) content{};
  leFileRead.read(reinterpret_cast<char *>(&content), sizeof(content));
  leFileRead.close();

  // check
  std::cout << "Written Payload: " << payload << std::endl;
  std::cout << "Read content   : " << std::dec << content << std::endl;
  std::cout << "Written hex    : " << std::hex << payload << std::endl;
  std::cout << "Read hex       : " << std::hex << content << std::endl;
```

Reply:
I was not able to find an alternative way to write a binary file. Although the generated file is interpreted as textfile, afterwards, I cannot tell if the "write mode" is binary, anyway.
In the end, I would prefer not to investigate this further, as I don't think I can figure this out fast enough to make it worth the effort.

Reply:
I'm ok with leaving the output in ASCII. However, write an issue that we (not necessarily you) should do this at some point. Also please include my code snippet from the comment above.
You can check the actual content of a file by looking at its hex representation. E.g. in vim via `:%!xxd`.
If you stay with ASCII I think you should remove the extra steps of calling `convertToAsciiString` because it actually does nothing.

Reply:
I will create an Issue. The convertToAsciiString is required for Paraview to be able to read the data. Don't ask me why, but it cannot interpret plain text, in this case.